packages/relay-kit/src/packs/safe-4337/Safe4337Pack.ts

```diff
@@ -220,14 +285,36 @@ export class Safe4337Pack extends RelayKitBasePack<{
       })
     }
 
-    const adjustEstimationData = await feeEstimator?.adjustEstimation?.({
-      bundlerUrl: this.#BUNDLER_URL,
-      entryPoint: this.#ENTRYPOINT_ADDRESS,
-      userOperation: safeOperation.toUserOperation()
-    })
+    const isPaymasterPresent = userOperation.paymasterAndData !== '0x'
+    const isDeplomentPresent = userOperation.initCode !== '0x'
 
-    if (adjustEstimationData) {
-      safeOperation.addEstimations(adjustEstimationData)
+    // adjustment only needed for deployments without paymaster
+    if (!isPaymasterPresent && isDeplomentPresent) {
+      const adjustEstimationData = await feeEstimator?.adjustEstimation?.({
+        bundlerUrl: this.#BUNDLER_URL,
+        entryPoint: this.#ENTRYPOINT_ADDRESS,
+        userOperation: safeOperation.toUserOperation()
+      })
+
+      if (adjustEstimationData) {
+        safeOperation.addEstimations(adjustEstimationData)
+      }
+    }
+
+    if (isPaymasterPresent) {
+      if (!this.#PAYMASTER_URL) {
+        throw new Error('No paymaster url provided')
+      }
+
+      const paymasterEstimation = await feeEstimator?.getPaymasterEstimation?.({
```

Comment:
There is a typo here and I suggest a different naming convention
```suggestion
    const usePaymaster = userOperation.paymasterAndData !== '0x'
    const hasSafeDeployment = userOperation.initCode !== '0x'

    // adjustment only needed for deployments without paymaster
    if (!usePaymaster && hasSafeDeployment) {
      const adjustEstimationData = await feeEstimator?.adjustEstimation?.({
        bundlerUrl: this.#BUNDLER_URL,
        entryPoint: this.#ENTRYPOINT_ADDRESS,
        userOperation: safeOperation.toUserOperation()
      })

      if (adjustEstimationData) {
        safeOperation.addEstimations(adjustEstimationData)
      }
    }

    if (usePaymaster) {
      if (!this.#PAYMASTER_URL) {
        throw new Error('No paymaster url provided')
      }

      const paymasterEstimation = await feeEstimator?.getPaymasterEstimation?.({
```

Reply:
done!